README.md

```diff
@@ -107,41 +108,111 @@ If you are attempting to do this on an OpenShift cluster, you will need to grant
 
-The secret should have the name: *crname*-postgres-configuration and
-should look like:
+#### External Database
 
-    apiVersion: v1
-    kind: Secret
-    metadata:
-      name: <crname>-postgres-configuration
-      namespace: <target namespace>
-    stringData:
-      host: <external ip or url resolvable by the cluster>
-      port: <external port, this usually defaults to 5432>
-      database: <desired database name>
-      username: <username to connect as>
-      password: <password to connect with>
-    type: Opaque
+In the scenario when one wants to rely on an external PostgreSQL instance, one needs to provide the necessary information to the operator.
 
-### Persistent storage for Postgres
+There are three ways one can specify those information. Those way are listed in order of priority in which they are treated.
 
-If you need to use a specific storage class for Postgres' storage, specify `tower_postgres_storage_class` in your AWX spec:
+* Providing a specifically crafted secret
 
-    ---
-    spec:
-      ...
-      tower_postgres_storage_class: fast-ssd
+One can create a specially formated secret within the namespace the AWX instance will be deployed into. This secret needs then to be specified as `tower_postgres_configuration_secret` variable in the spec requirements.
+
+```yaml
+spec:
+  external_database: true
+  tower_postgres_configuration_secret: my-awx-external-db-secret
+```
+
+The secret format is the following:
+
+```yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: <name-of-the-secret>
+  namespace: <namespace>
+stringData:
+  password: <password>
+  username: <username>
+  database: <database>
+  port: <port>
+  host: <host>
+```
+
+* Providing variables (not recommended)
+
+If one does not wish to use secret, one can specify the following variables in the spec requirements.
+
+| Variable                | Descritpion                       | Default Value |
+| ----------------------- | --------------------------------- | ------------- |
+| tower_postgres_user     | Username to connect to PostgreSQL |               |
+| tower_postgres_pass     | Password to connect to PostgreSQL |               |
+| tower_postgres_database | Database name to connect to       |               |
+| tower_postgres_host     | Host to connect to PostgreSQL     |               |
+| tower_postgres_port     | Port to connect to PostgreSQL     | 5432          |
 
-If it's not specified, Postgres will store it's data on a volume using the default storage class for your cluster.
+The following is an example of an AWX spec section that relie on those variables.
+
+```yaml
+spec:
+  external_database: true
+  tower_postgres_user: awx
+  tower_postgres_pass: awx
```

Comment:
⚠️ This is not secure. We should _never_ instruct the user to provide sensitive information in anything except a `Secret`.